src/util.rs

```diff
@@ -218,6 +222,16 @@ pub(crate) async fn start_accepting_connections(
             }
         }
     });
+    debug_assert_ne!(resolved_addr.port(), 0);
+    *port = resolved_addr.port();
+    let resolved_host: Host = resolved_addr.ip().into();
+    if let Host::Ipv4(ip) = host {
+        debug_assert!(ip == &mut resolved_addr.ip());
```

Comment:
I was surprised to find that when given the unspecified address (0.0.0.0 or ::), `local_addr()` did not resolve it to a particular ip addr like `127.0.0.1`. Is that expected behavior?